sdk/storage/azure-storage-file/src/test/java/com/azure/storage/file/spock/ShareAPITests.groovy

```diff
@@ -0,0 +1,350 @@
+package com.azure.storage.file.spock
+
+import com.azure.core.exception.HttpResponseException
+import com.azure.storage.common.credentials.SharedKeyCredential
+import com.azure.storage.file.ShareClientBuilder
+import com.azure.storage.file.models.FileHTTPHeaders
+import com.azure.storage.file.models.StorageErrorCode
+import com.azure.storage.file.models.StorageErrorException
+import spock.lang.Unroll
+
+import java.time.LocalDateTime
+import java.time.OffsetDateTime
+import java.time.ZoneOffset
+
+import static org.junit.Assert.assertEquals
+import static org.junit.Assert.assertNotNull
+
+class ShareAPITests extends APISpec {
+    def primaryShareClient
+    def shareName
+    static def testMetadata
+
+    def setup() {
+        shareName = testResourceName.randomName("share", 16)
+        primaryFileServiceClient = fileServiceBuilderHelper(interceptorManager).buildClient()
+        primaryShareClient = primaryFileServiceClient.getShareClient(shareName)
+        testMetadata = Collections.singletonMap("testmetadata", "value")
+    }
+
+    def "Get share URL from share client"() {
+        given:
+        def accoutName = SharedKeyCredential.fromConnectionString(connectionString).accountName()
+        def expectURL = String.format("https://%s.file.core.windows.net", accoutName)
+        when:
+        def shareURL = primaryShareClient.getShareUrl().toString()
+        then:
+        expectURL.equals(shareURL)
+    }
+
+    def "Get root directory client does not create a directory from share client"() {
+        given:
+        primaryShareClient.create()
+        def directoryClient = primaryShareClient.getRootDirectoryClient()
+        when:
+        def getPropertiesResponse = directoryClient.getProperties()
+        then:
+        FileTestHelper.assertResponseStatusCode(getPropertiesResponse, 200)
+        assertNotNull(getPropertiesResponse.value().eTag())
+    }
+
+    def "Get file client does not create a file from share client"() {
+        given:
+        primaryShareClient.create()
+        when:
+        def fileClient = primaryShareClient.getFileClient("testFile")
+        fileClient.getProperties()
+        then:
+        assertNotNull(fileClient)
+        def e = thrown(HttpResponseException)
+        assertEquals(e.response().statusCode(), 404)
+    }
+
+    def "Create share from share client"() {
+        expect:
+        FileTestHelper.assertResponseStatusCode(primaryShareClient.create(), 201)
+    }
+
+    @Unroll
+    def "Create share with args from share client"() {
+        expect:
+        FileTestHelper.assertResponseStatusCode(primaryShareClient.create(metadata, quota), 201)
+        where:
+        metadata     | quota
+        null         | null
```

Comment:
Doesn't this case cover the test above? Let's merge them and just have a single "Create share from share client" test.

Reply:
This one is testing API with overload. The above one is testing min overload. Slightly different.